CMakeExt/CompilerFlags.cmake

```diff
@@ -19,7 +19,7 @@ find_package(OpenMP)
 #  | -Weffc++                 | Spurious false positives                  |
 #  '--------------------------'-------------------------------------------'
 
-set (DART_C_STD_PREFERED "99")
+set (DART_C_STD_PREFERED "11")
```

Comment:
As we only require a c99 capable compiler, this is not possible. We have to guard this similar to the implementation in PR #528 . Will fix it there.